crates/emitter/src/scope_notes.rs

```diff
@@ -62,6 +62,27 @@ impl ScopeNoteList {
         ScopeNoteIndex::new(note_index)
     }
 
+    fn get_scope_hole_gcthing_index(&self, scope_note_index: &ScopeNoteIndex) -> GCThingIndex {
+        self.notes
+            .get(scope_note_index.index)
+            .expect("Scope note should exist")
+            .index
+    }
+
+    pub fn enter_scope_hole(
+        &mut self,
+        maybe_enclosing_scope_note_index: &Option<ScopeNoteIndex>,
+        body_index: GCThingIndex,
+        offset: BytecodeOffset,
+    ) -> ScopeNoteIndex {
```

Comment:
I think it's better separating this into 2 methods,
one for the case it has scope note, and one for body scope.
so that the parameter of the method is clear and has no duplicate.

Reply:
Once I did this i realized that the only way this differs from enter_scope is the gcindex retrieval. that might be better handled in the instruction writer...